File: app/src/main/java/com/noalecohen/dispatcher/MainActivity.java
package com.noalecohen.dispatcher;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import com.google.android.material.tabs.TabLayout;
import com.noalecohen.dispatcher.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;
    FragmentManager fragmentManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        bindingView();
        setBottomNavigation();
    }

    private void bindingView(){
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        fragmentManager = getSupportFragmentManager();
    }

    private void setBottomNavigation() {
        binding.homePageBottomNav.getTabAt(1).select();
        binding.homePageBottomNav.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                switch (tab.getPosition()){
                    case 0:
                        fragmentManager.beginTransaction().replace(binding.homePageFrameContent.getId(), new AccountFragment()).commit();
                        break;
                    case 1:
                        fragmentManager.beginTransaction().replace(binding.homePageFrameContent.getId(), new HomeFragment()).commit();
                        break;
                    case 2:
                        fragmentManager.beginTransaction().replace(binding.homePageFrameContent.getId(), new FavoritesFragment()).commit();
                        break;
                    default:
                        break;
                }
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });
    }
}

File: app/src/main/java/com/noalecohen/dispatcher/MainActivity.kt
package com.noalecohen.dispatcher

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import com.google.android.material.tabs.TabLayout
import com.noalecohen.dispatcher.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    var fragmentManager: FragmentManager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        bindingView()
        setBottomNavigation(savedInstanceState)
    }

    private fun bindingView() {
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        fragmentManager = supportFragmentManager
    }

    private fun setBottomNavigation(savedInstanceState: Bundle?) {
        var tabIndex = 1
        if (savedInstanceState != null) {
            tabIndex = savedInstanceState.get("index") as Int
        }
        binding.homePageBottomNav.getTabAt(tabIndex)?.select()

        binding.homePageBottomNav.addOnTabSelectedListener(object :
            TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                when (tab.position) {
                    0 -> fragmentManager!!.beginTransaction()
                        .replace(binding.homePageFrameContent.id, AccountFragment()).commit()
                    1 -> fragmentManager!!.beginTransaction()
                        .replace(binding.homePageFrameContent.id, HomeFragment()).commit()
                    2 -> fragmentManager!!.beginTransaction()
                        .replace(binding.homePageFrameContent.id, FavoritesFragment()).commit()
                    else -> {
                        Log.d("TAG", "Tab not found")
                    }
                }
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {}
            override fun onTabReselected(tab: TabLayout.Tab) {}
        })
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        val index = binding.homePageBottomNav.selectedTabPosition
        outState.putInt("index", index)
    }
}